File: 2023-06-28-hmac/action-in-blog/src/main/java/action/in/blog/TodoController.java
package action.in.blog;

import lombok.Builder;
import org.springframework.web.bind.annotation.*;

import java.util.Arrays;
import java.util.List;

record Todo(
        long id,
        String content
) {
    @Builder
    public Todo {
    }
}

@RestController
class TodoController {

    @GetMapping("/todos")
    public List<Todo> getPosts() {
        return Arrays.asList(
                new Todo(1000, "Hello"),
                new Todo(1001, "World"),
                new Todo(1002, "Study"),
                new Todo(1003, "Java")
        );
    }

    @DeleteMapping("/todos/{id}")
    public long deleteTodo(@PathVariable long id) {
        return id;
    }

    @PostMapping("/todos")
    public Todo createTodo(@RequestBody Todo todo) {
        return Todo.builder()
                .id(2000)
                .content(todo.content())
                .build();
    }

    @PutMapping("/todos")
    public Todo updateTodo(Todo todo) {
        return Todo.builder()
                .id(todo.id())
                .content(todo.content())
                .build();
    }
}

File: 2024-02-07-timezone-problem-of-jvm-application/action-in-blog/src/main/kotlin/blog/in/action/controller/TodoController.kt
package blog.`in`.action.controller

import blog.`in`.action.domain.Todo
import blog.`in`.action.repository.TodoRepository
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class TodoController(private val todoRepository: TodoRepository) {

    @GetMapping("/todos")
    fun todos(): List<Todo> = todoRepository.findAll().map { Todo.of(it) }.toList()
}